library/Vanilla/Embeds/GettyEmbed.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+/**
+ * Getty Embed.
+ */
+class GettyEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $domains = ['gettyimages.ca', 'gty.im', 'gettyimages.com'];
+
+    /**
+     * GettyEmbed constructor.
+     */
+    public function __construct() {
+        parent::__construct('getty', 'image');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function matchUrl(string $url) {
+        $data = [];
+
+        if ($this->isNetworkEnabled()) {
+            preg_match(
+                '/https?:\/\/www\.gettyimages\.(com|ca).*?(?<postID>\d+)(?!.*\d)/i',
+                $url,
+                $post
+            );
+            if (!$post) {
+                throw new Exception('Unable to get post ID.', 400);
+            }
+
+            $oembedData = $this->oembed("http://embed.gettyimages.com/oembed?url=http://gty.im/".$post['postID']);
+
+            if ($oembedData) {
+                $data = $oembedData;
+            }
+
+            if (array_key_exists('html', $data)) {
+                $data['attributes'] = $this->parseResponseHtml($data['html']);
+            }
+
+            if ($post) {
+                $data['attributes']['postID'] = $post['postID'];
```

Comment:
We start with an empty `$data` array. __*If*__ we get back some good oEmbed data, we set the result to `$data`. __*If*__ `$data` has an "html" attribute, we parse it and populate its "attributes" field with the result.

With the way things are structured, neither of those things are guaranteed to have happened by the time you reach this line. You're potentially referencing a non-existent key ("attributes") and then treating that potentially non-existent value as an array, referencing another key ("postID").